### ISSUE TITLE
Unable to pull latest image of kaniko-executor on ppc64le platform

### ISSUE THREAD

**Issue by AaruniAggarwal** [GoogleContainerTools/kaniko#1904]
While pulling the latest image of kaniko-executor on ppc64le, I am getting the following error : 

 `Error: choosing an image from manifest list docker://gcr.io/kaniko-project/executor:v1.7.0: no image found in the manifest list for architecture ppc64le, variant "", OS Linux 

ie. 

```
# docker pull gcr.io/kaniko-project/executor:v1.7.0
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
Trying to pull gcr.io/kaniko-project/executor:v1.7.0...
Error: choosing an image from manifest list docker://gcr.io/kaniko-project/executor:v1.7.0: no image found in manifest list for architecture ppc64le, variant "", OS linux

```

**Comment by AaruniAggarwal** [comment]
@tejal29 , I found out that you have removed ppc64le support here - https://github.com/GoogleContainerTools/kaniko/pull/1784
Is there any specific reason for doing so ? 

Thanks

**Comment by imjasonh** [comment]
That image was not built for that architecture at v1.7.0: https://github.com/GoogleContainerTools/kaniko/blob/v1.7.0/.github/workflows/release.yaml#L60

It's also not configured to build that arch currently, so if you'd like to have it in the future we'll have to add it.

**Comment by imjasonh** [comment]
I'm not sure why it was removed, if you propose adding it here I'll approve and merge: https://github.com/GoogleContainerTools/kaniko/blob/main/.github/workflows/images.yaml#L36

**Comment by AaruniAggarwal** [comment]
@imjasonh, I made the following changes : 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# git diff
diff --git a/.github/workflows/images.yaml b/.github/workflows/images.yaml
index 09e3cbc2..6c1a9b9f 100644
--- a/.github/workflows/images.yaml
+++ b/.github/workflows/images.yaml
@@ -33,14 +33,14 @@ jobs:
         include:
         - image: executor
           dockerfile: ./deploy/Dockerfile
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/executor
           tag: ${{ github.sha }}
           release-tag: latest
 
         - image: executor-debug
           dockerfile: ./deploy/Dockerfile_debug
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/executor
           tag: ${{ github.sha }}-debug
           release_tag: debug
@@ -54,7 +54,7 @@ jobs:
 
         - image: warmer
           dockerfile: ./deploy/Dockerfile_warmer
-          platforms: linux/amd64,linux/arm64,linux/s390x
+          platforms: linux/amd64,linux/arm64,linux/s390x,linux/ppc64le
           image-name: gcr.io/kaniko-project/warmer
           tag: ${{ github.sha }}
           release-tag: latest
```
However while running `make images` command, I am getting following error :  

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# make images
docker build --platform linux/ppc64le --build-arg=GOARCH=ppc64le -t gcr.io/kaniko-project/executor:latest -f deploy/Dockerfile .
Emulate Docker CLI using podman. Create /etc/containers/nodocker to quiet msg.
WARN[0000] missing "TARGETARCH" build argument. Try adding "--build-arg TARGETARCH=<VALUE>" to the command line 
[1/3] STEP 1/12: FROM golang:1.17
? Please select an image: 
✔ docker.io/library/golang:1.17
Trying to pull docker.io/library/golang:1.17...
Getting image source signatures
Copying blob 9ecf28457895 done  
Copying blob 3666fa80bd5b done  
Copying blob c0d4b2a71bc2 done  
Copying blob 8199fa873cad done  
Copying blob 7c968dbebedd done  
Copying blob 91c8a760808c done  
Copying blob 7651f3f70ff5 done  
Copying config f06e318d69 done  
Writing manifest to image destination
Storing signatures
[1/3] STEP 2/12: WORKDIR /src
--> 4d0298feae3
[1/3] STEP 3/12: ARG TARGETARCH
--> dd91ae30e4e
[1/3] STEP 4/12: ENV GOARCH=$TARGETARCH
--> 1872e3298b2
[1/3] STEP 5/12: ENV CGO_ENABLED=0
--> c0135c7c07a
[1/3] STEP 6/12: ENV GOBIN=/usr/local/bin
--> 11a2daa0123
[1/3] STEP 7/12: RUN go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8
go: downloading github.com/GoogleCloudPlatform/docker-credential-gcr v1.5.1-0.20210909233216-4cdd60d0f2d8
go: downloading github.com/google/subcommands v1.2.0
go: downloading github.com/docker/cli v20.10.8+incompatible
go: downloading github.com/docker/docker-credential-helpers v0.6.4
go: downloading github.com/toqueteos/webbrowser v1.2.0
go: downloading golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f
go: downloading cloud.google.com/go v0.65.0
go: downloading golang.org/x/net v0.0.0-20210405180319-a5a99cb37ef4
go: downloading github.com/docker/docker v20.10.8+incompatible
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/sirupsen/logrus v1.8.1
go: downloading golang.org/x/sys v0.0.0-20210510120138-977fb7262007
--> a0c8dc24b57
[1/3] STEP 8/12: RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login@69c85dc22db6511932bbf119e1a0cc5c90c69a7f # v0.6.0
go: downloading github.com/awslabs/amazon-ecr-credential-helper/ecr-login v0.0.0-20211215200129-69c85dc22db6
go: downloading github.com/awslabs/amazon-ecr-credential-helper v0.6.0
go: downloading github.com/docker/docker-credential-helpers v0.6.3
go: downloading github.com/aws/aws-sdk-go-v2 v1.7.1
go: downloading github.com/aws/aws-sdk-go-v2/config v1.5.0
go: downloading github.com/aws/aws-sdk-go-v2/service/ecr v1.4.1
go: downloading github.com/aws/aws-sdk-go-v2/service/ecrpublic v1.4.1
go: downloading github.com/aws/smithy-go v1.6.0
go: downloading github.com/sirupsen/logrus v1.4.2
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/aws/aws-sdk-go-v2/credentials v1.3.1
go: downloading github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.3.0
go: downloading github.com/aws/aws-sdk-go-v2/internal/ini v1.1.1
go: downloading github.com/aws/aws-sdk-go-v2/service/sso v1.3.1
go: downloading github.com/aws/aws-sdk-go-v2/service/sts v1.6.0
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading golang.org/x/sys v0.0.0-20210423082822-04245dca01da
go: downloading github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.2.1
--> 47c42c07ea6
[1/3] STEP 9/12: RUN go install github.com/chrismellard/docker-credential-acr-env@09e2b5a8ac86c3ec347b2473e42b34367d8fa419
go: downloading github.com/chrismellard/docker-credential-acr-env v0.0.0-20210203204924-09e2b5a8ac86
go: downloading github.com/spf13/viper v1.7.1
go: downloading github.com/spf13/cobra v1.0.0
go: downloading github.com/Azure/go-autorest/autorest/azure/auth v0.5.2
go: downloading github.com/Azure/azure-sdk-for-go v46.4.0+incompatible
go: downloading github.com/Azure/go-autorest/autorest v0.11.8
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.4
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/fsnotify/fsnotify v1.4.9
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/magiconair/properties v1.8.4
go: downloading github.com/mitchellh/mapstructure v1.3.3
go: downloading github.com/pelletier/go-toml v1.8.1
go: downloading github.com/spf13/afero v1.4.1
go: downloading github.com/spf13/cast v1.3.1
go: downloading github.com/spf13/jwalterweatherman v1.1.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/subosito/gotenv v1.2.0
go: downloading gopkg.in/ini.v1 v1.62.0
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.1
go: downloading github.com/dimchansky/utfbom v1.1.0
go: downloading github.com/Azure/go-autorest/logger v0.2.0
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading golang.org/x/crypto v0.0.0-20201002170205-7f63de1d35b0
go: downloading golang.org/x/sys v0.0.0-20201005172224-997123666555
go: downloading golang.org/x/text v0.3.3
--> 5d68566548e
[1/3] STEP 10/12: RUN mkdir -p /kaniko/.docker
--> cba0914d42f
[1/3] STEP 11/12: COPY . .
--> 176af4ac956
[1/3] STEP 12/12: RUN     --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg     make GOARCH=$TARGETARCH
[2/3] STEP 1/2: FROM debian:buster-slim AS certs
Resolved "debian" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/debian:buster-slim...
Error: error building at STEP "RUN --mount=type=cache,target=/root/.cache/go-build --mount=type=cache,target=/go/pkg make GOARCH=$TARGETARCH": error resolving mountpoints for container "d0cd790f174de02a97b654c87fd068d7a924bdc407f23834a61448081b05f0c8": invalid filesystem type "cache"
make: *** [Makefile:97: images] Error 125
```
Not sure, why ?

**Comment by imjasonh** [comment]
```
Error: error building at STEP "RUN --mount=type=cache,target=/root/.cache/go-build --mount=type=cache,target=/go/pkg make GOARCH=$TARGETARCH": error resolving mountpoints for container "d0cd790f174de02a97b654c87fd068d7a924bdc407f23834a61448081b05f0c8": invalid filesystem type "cache"
```

I believe this means you need to set `DOCKER_BUILDKIT=1` to be able to run `make images`. This behavior was introduced in https://github.com/GoogleContainerTools/kaniko/pull/1853, we should probably just set it ourselves inside `make images`. Sorry about that.

The diff looks good though, can you send a PR?

**Comment by AaruniAggarwal** [comment]
although , when i checked `podman images` , I got kaniko-executor image for ppc64le : 

```
[root@rdr-odf-upg2-syd05-bastion-0 ~]# podman images |grep executor
gcr.io/kaniko-project/executor               latest              15bfb919360b  39 minutes ago     61.9 MB
```

inspect of an image: 

```
[root@rdr-odf-upg2-syd05-bastion-0 ~]# podman inspect gcr.io/kaniko-project/executor:latest
[
    {
        "Id": "15bfb919360b19ec17d654f260ca572e99048e2612c7e85f109b7072a9b58979",
        "Digest": "sha256:92f8ee6b742b8a23e5b7dc6a0e36819abafe98ff5f278f1a50e871cce528234b",
        "RepoTags": [
            "gcr.io/kaniko-project/executor:latest"
        ],
        "RepoDigests": [
            "gcr.io/kaniko-project/executor@sha256:92f8ee6b742b8a23e5b7dc6a0e36819abafe98ff5f278f1a50e871cce528234b"
        ],
        "Parent": "f3285067570147a50dc284f811e50893912febda326c40dd3cdc54a55e70f055",
        "Comment": "",
        "Created": "2022-01-31T18:14:27.760603911Z",
        "Config": {
            "Env": [
                "PATH=/usr/local/bin:/kaniko",
                "HOME=/root",
                "USER=root",
                "SSL_CERT_DIR=/kaniko/ssl/certs",
                "DOCKER_CONFIG=/kaniko/.docker/",
                "DOCKER_CREDENTIAL_GCR_CONFIG=/kaniko/.config/gcloud/docker_credential_gcr_config.json"
            ],
            "Entrypoint": [
                "/kaniko/executor"
            ],
            "WorkingDir": "/workspace",
            "Labels": {
                "io.buildah.version": "1.22.3"
            }
        },
        "Version": "",
        "Author": "",
        "Architecture": "ppc64le",
        "Os": "linux",
        "Size": 61931190,
        "VirtualSize": 61931190,
        "GraphDriver": {
            "Name": "overlay",
            "Data": {
                "LowerDir": "/var/lib/containers/storage/overlay/2d624453a3c6d61007d35223afc7f08468f9264cf7711cc86440a9ecdb657159/diff:/var/lib/containers/storage/overlay/08983fe4c25c93e4c0a4b051a52898f9a54a4fdb835218820941d4f088e04478/diff:/var/lib/containers/storage/overlay/0cb718d18fa34972227dc7055bceebe56c32121a7144f153528a7ff9bfd0a457/diff:/var/lib/containers/storage/overlay/b21304efddf987c0fc95233bbb14b68b42798327de53a9d0c63318cd88bf2134/diff:/var/lib/containers/storage/overlay/13e1e9688742cdec77a5fb830edcd46f7f47346413f93d4139c162307f81c8f4/diff:/var/lib/containers/storage/overlay/d5532cc681f1321fc999a48ec2aea2fa44c90bd010d5bb0d03c9801d9086765d/diff",
                "UpperDir": "/var/lib/containers/storage/overlay/1fa83132559b6c39663c45a6b18f5bd9510b05579eede2f130dfb4567dab9bc1/diff",
                "WorkDir": "/var/lib/containers/storage/overlay/1fa83132559b6c39663c45a6b18f5bd9510b05579eede2f130dfb4567dab9bc1/work"
            }
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:d5532cc681f1321fc999a48ec2aea2fa44c90bd010d5bb0d03c9801d9086765d",
                "sha256:212d109b8f5514c76cddde59be700973721673a909fb7c090d3cae3e479f3bb3",
                "sha256:c958931fc3f7e77e128fd0901f0ad2f40f3e3c5e3027a17f69b219a011fbc8ec",
                "sha256:b42ea2fc7e3e8611823aeffd866dd29b89abcb5d59015f5819d049d52f2f7783",
                "sha256:9799170614fbf26ba8acaa52cc1c5a4b699417d525277bf0d174a11fe0083824",
                "sha256:4d4892e2e803ebcff73e6f342261ee2b2e9b568a7fcdc398a82f22d13a7e2622",
                "sha256:01c2e7c258994c0939c49c5ab102c5f507436b05416919e50bad11c31e799d31"
            ]
        },
        "Labels": {
            "io.buildah.version": "1.22.3"
        },
        "Annotations": {},
        "ManifestType": "application/vnd.oci.image.manifest.v1+json",
        "User": "",
        "History": [
            {
                "created": "2022-01-31T18:14:19.875360466Z",
                "created_by": "/bin/sh -c #(nop) COPY file:5150c6c6cb8200bf1808670d229565c2346d196e19a404da0b2897b3180e6a12 in /kaniko/executor "
            },
            {
                "created": "2022-01-31T18:14:22.000267698Z",
                "created_by": "/bin/sh -c #(nop) COPY file:a80752a0fca2b627b5333a02cb698517d4e41711456b92bcbfdbe018d32ef8d5 in /kaniko/docker-credential-gcr "
            },
            {
                "created": "2022-01-31T18:14:23.000320433Z",
                "created_by": "/bin/sh -c #(nop) COPY file:b8806b314be425c1e553112bf2e087645065da6270953645e0d746516be9879f in /kaniko/docker-credential-ecr-login "
            },
            {
                "created": "2022-01-31T18:14:24.000261959Z",
                "created_by": "/bin/sh -c #(nop) COPY file:9cd723207ac9bc5ab83e487acf4ae1345fe0ae4920bb3cd16bfa9581cd0f10c0 in /kaniko/docker-credential-acr-env "
            },
            {
                "created": "2022-01-31T18:14:25.000065607Z",
                "created_by": "/bin/sh -c #(nop) COPY file:46647a8375730b2c106b4218ada194b617dc95ea94dd2cbb5d53ebb790380914 in /kaniko/ssl/certs/ "
            },
            {
                "created": "2022-01-31T18:14:26.000134531Z",
                "created_by": "/bin/sh -c #(nop) COPY dir:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef in /kaniko/.docker "
            },
            {
                "created": "2022-01-31T18:14:27.000620672Z",
                "created_by": "/bin/sh -c #(nop) COPY file:72d1949654e4b2fc2ad4da33210c8d55dcf313656ec30a9eb95e3cd297ddf7fe in /etc/nsswitch.conf "
            },
            {
                "created": "2022-01-31T18:14:27.248124799Z",
                "created_by": "/bin/sh -c #(nop) ENV HOME /root",
                "empty_layer": true
            },
            {
                "created": "2022-01-31T18:14:27.297297724Z",
                "created_by": "/bin/sh -c #(nop) ENV USER root",
                "empty_layer": true
            },
            {
                "created": "2022-01-31T18:14:27.342225247Z",
                "created_by": "/bin/sh -c #(nop) ENV PATH /usr/local/bin:/kaniko",
                "empty_layer": true
            },
            {
                "created": "2022-01-31T18:14:27.395931198Z",
                "created_by": "/bin/sh -c #(nop) ENV SSL_CERT_DIR=/kaniko/ssl/certs",
                "empty_layer": true
            },
            {
                "created": "2022-01-31T18:14:27.451369908Z",
                "created_by": "/bin/sh -c #(nop) ENV DOCKER_CONFIG /kaniko/.docker/",
                "empty_layer": true
            },
            {
                "created": "2022-01-31T18:14:27.498540458Z",
                "created_by": "/bin/sh -c #(nop) ENV DOCKER_CREDENTIAL_GCR_CONFIG /kaniko/.config/gcloud/docker_credential_gcr_config.json",
                "empty_layer": true
            },
            {
                "created": "2022-01-31T18:14:27.628226549Z",
                "created_by": "/bin/sh -c #(nop) WORKDIR /workspace",
                "empty_layer": true
            },
            {
                "created": "2022-01-31T18:14:27.761653842Z",
                "created_by": "/bin/sh -c #(nop) ENTRYPOINT [\"/kaniko/executor\"]",
                "empty_layer": true
            }
        ],
        "NamesHistory": [
            "gcr.io/kaniko-project/executor:latest"
        ]
    }
]
```

sure  Jason, I will send a PR .
Thanks !!!

**Comment by AaruniAggarwal** [comment]
I manually added `DOCKER_BUILDKIT=1` in Makefile . But it is still failing with the same error. 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# git diff Makefile
diff --git a/Makefile b/Makefile
index 1a3b87c5..3e5a17e8 100644
--- a/Makefile
+++ b/Makefile
@@ -93,6 +93,7 @@ k8s-executor-build-push:
        docker push $(REGISTRY)/executor:latest
 
 .PHONY: images
+images: DOCKER_BUILDKIT=1
 images:
        docker build ${BUILD_ARG} --build-arg=GOARCH=$(GOARCH) -t $(REGISTRY)/executor:latest -f deploy/Dockerfile .
        docker build ${BUILD_ARG} --build-arg=GOARCH=$(GOARCH) -t $(REGISTRY)/executor:debug -f deploy/Dockerfile_debug .
```

error : 

```
Error: error building at STEP "RUN --mount=type=cache,target=/root/.cache/go-build --mount=type=cache,target=/go/pkg make GOARCH=$TARGETARCH": error resolving mountpoints for container "1a3cdd0a72265cedec09ed66c09a0ee736e6f7fdafac637308114ec53d8d4b2e": invalid filesystem type "cache"
make: *** [Makefile:98: images] Error 125
```

**Comment by AaruniAggarwal** [comment]
Jason, it seems I found the reason for the above error. 
According to the following documentation, it states podman doesn't support mount type=cache. (below heading: `Caching mounts: not supported by Podman` )

https://pythonspeed.com/articles/podman-buildkit/

**Comment by AaruniAggarwal** [comment]
Jason, with docker version : 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# docker version
Client: Docker Engine - Community
 Version:           v20.10.12
 API version:       1.41
 Go version:        go1.16.12
 Git commit:        e91ed57
 Built:             Tue Dec 14 10:12:12 2021
 OS/Arch:           linux/ppc64le
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          v20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.12
  Git commit:       459d0df
  Built:            Tue Dec 14 10:10:07 2021
  OS/Arch:          linux/ppc64le
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

```

and go version -> `go version go1.17.6 linux/ppc64le'

image build proceeded, but it is still failing with new error : 

```
[root@rdr-odf-upg2-syd05-bastion-0 kaniko]# make images
DOCKER_BUILDKIT=1 docker build  --build-arg=GOARCH=ppc64le -t gcr.io/kaniko-project/executor:latest -f deploy/Dockerfile .
[+] Building 161.8s (10/23)                                                                                                                                                                                 
 => [internal] load build definition from Dockerfile                                                                                                                                                   0.0s
 => => transferring dockerfile: 92B                                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/debian:buster-slim                                                                                                                                  1.8s
 => [internal] load metadata for docker.io/library/golang:1.17                                                                                                                                         1.8s
 => [stage-0 1/8] FROM docker.io/library/golang:1.17@sha256:301609ebecc0ec4cd3174294220a4d9c92aab9015b3a2958297d7663aac627a1                                                                           0.0s
 => [internal] load build context                                                                                                                                                                      0.3s
 => => transferring context: 784.92kB                                                                                                                                                                  0.3s
 => [certs 1/2] FROM docker.io/library/debian:buster-slim@sha256:f6e5cbc7eaaa232ae1db675d83eabfffdabeb9054515c15c2fb510da6bc618a7                                                                      0.0s
 => CACHED [stage-0 2/8] WORKDIR /src                                                                                                                                                                  0.0s
 => ERROR [stage-0 3/8] RUN go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8                                                                 159.8s
 => CACHED [certs 2/2] RUN   apt update &&   apt install -y ca-certificates &&   cat /etc/ssl/certs/* > /ca-certificates.crt                                                                           0.0s
------                                                                                                                                                                                                      
 > [stage-0 3/8] RUN go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8:                                                                              
#7 6.526 go: downloading github.com/GoogleCloudPlatform/docker-credential-gcr v1.5.1-0.20210909233216-4cdd60d0f2d8                                                                                          
#7 159.5 go install: github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8: github.com/GoogleCloudPlatform/docker-credential-gcr@v1.5.1-0.20210909233216-4cdd60d0f2d8: read tcp 172.17.0.2:33226->142.251.46.177:443: read: connection reset by peer
------
executor failed running [/bin/sh -c go install github.com/GoogleCloudPlatform/docker-credential-gcr@4cdd60d0f2d8a69bc70933f4d7718f9c4e956ff8]: exit code: 1
make: *** [Makefile:97: images] Error 1
```

Not sure what I am missing now ?

**Comment by imjasonh** [comment]
That seems like a transient network error, are you seeing that reliably? I'm not able to reproduce.

**Comment by AaruniAggarwal** [comment]
Yes, for the past 3-4 times, I am getting the same error. 

Is it fine, if I still submit the PR with the changes, or do I need to wait for it to build successfully?

**Comment by imjasonh** [comment]
Yeah just send a PR and if it doesn't fail on CI we'll merge it.

Thanks for working through all this!

**Comment by AaruniAggarwal** [comment]
Here is the PR: https://github.com/GoogleContainerTools/kaniko/pull/1908

Thanks, Jason for all your help.

**Comment by AaruniAggarwal** [comment]
As #1906 is still open, not sure whether CI will pass or not.

**Comment by AaruniAggarwal** [comment]
One of the check failed. [Build images / build-images (executor-debug) (pull_request) ] because of following error : 
```
#45 [stage-2 12/13] RUN ["/busybox/mkdir", "-p", "/bin"]
#45 0.072 qemu-ppc64le: Could not open '/lib64/ld64.so.2': No such file or directory
#45 ERROR: process "/busybox/mkdir -p /bin" did not complete successfully: exit code: 255
------
 > [stage-2 12/13] RUN ["/busybox/mkdir", "-p", "/bin"]:
#45 0.072 qemu-ppc64le: Could not open '/lib64/ld64.so.2': No such file or directory
------
Dockerfile_debug:78
--------------------
  76 |     ENV DOCKER_CREDENTIAL_GCR_CONFIG /kaniko/.config/gcloud/docker_credential_gcr_config.json
  77 |     WORKDIR /workspace
  78 | >>> RUN ["/busybox/mkdir", "-p", "/bin"]
  79 |     RUN ["/busybox/ln", "-s", "/busybox/sh", "/bin/sh"]
  80 |     ENTRYPOINT ["/kaniko/executor"]
--------------------
error: failed to solve: process "/busybox/mkdir -p /bin" did not complete successfully: exit code: 255
Error: buildx failed with: error: failed to solve: process "/busybox/mkdir -p /bin" did not complete successfully: exit code: 255
```

**Comment by imjasonh** [comment]
Sounds like there might be an issue with our assumptions about the busybox, being unsatisfied by the ppc64le image.

If you don't need the `:debug` variant, would you be okay omitting it for the debug image for now? We can add a comment or open a new issue to describe why it's missing in case someone in the future wants to figure it out to support it. That would unblock you at least.

**Comment by AaruniAggarwal** [comment]
Yes Jason, It sounds good.  Will remove `linux/ppc64le` for debug image as we only need executor image at the moment. 
Thanks, Jason

**Comment by AaruniAggarwal** [comment]
I have updated PR #1908 .

**Comment by AaruniAggarwal** [comment]
Jason, Could you please review PR #1908 

Thanks.
